runtime/ts/storage/storage-provider-base.ts

```diff
@@ -58,15 +60,19 @@ export class StorageProviderBase {
     return this._type;
   }
   // TODO: add 'once' which returns a promise.
-  on(kind, callback, target): void {
+  on(kindStr: string, callback: Callback, target): void {
     assert(target !== undefined, 'must provide a target to register a storage event handler');
+    const kind: EventKind = EventKind[kindStr];
```

Comment:
This doesn't really add anything useful - leaving the Event enum as string values only keeps the code more readable and eliminates the need to do lookups like this.

Reply:
Unless we get rid of the enum throughout this is needed for type safety.

Would you like me to change:
```
  private listeners: Map<EventKind, Map<Callback, {target: {}}>>;
```

to 

```
  private listeners: Map<string, Map<Callback, {target: {}}>>;
```

instead?

Reply:
I'll propose a better solution for this in a follow up PR.